doc/developers/demo.rst

```diff
@@ -1,76 +1,127 @@
 
-    $ cd build
-    $ ctest -VV -R e2e_scenario -N
+The following command can be run from your ``CCF/build`` directory to create and run a test network:
+
+.. code-block:: bash
 
+    $ ../start_test_network.sh --package ./liblogging.enclave.so.signed 
+    Setting up Python environment...
     ...
-    42: Test command: /usr/bin/unbuffer "python3" "/data/src/CCF/tests/e2e_scenarios.py" "-b" "." "--label" "e2e_scenario" "-l" "info" "-g" "/data/src/CCF/src/runtime_config/gov.lua" "--scenario" "/data/src/CCF/tests/simple_logging_scenario.json"
-    42: Environment variables:
-    42:  PYTHONPATH=/data/src/CCF/tests
+    Python environment successfully setup
+    [2020-07-17 15:27:41.759] Starting 3 CCF nodes...
+    [2020-07-17 15:27:46.643] Started CCF network with the following nodes:
+    [2020-07-17 15:27:46.643]   Node [ 0] = 127.251.192.205:36981
+    [2020-07-17 15:27:46.643]   Node [ 1] = 127.98.174.190:36795
+    [2020-07-17 15:27:46.643]   Node [ 2] = 127.113.40.231:33227
+    [2020-07-17 15:27:46.644] You can now issue business transactions to the ./liblogging.enclave.so.signed application.
+    [2020-07-17 15:27:46.644] Certificates have been copied to /data/src/CCF/build/workspace/test_network_common
+    [2020-07-17 15:27:46.644] See https://microsoft.github.io/CCF/users/issue_commands.html for more information.
+    [2020-07-17 15:27:46.644] Press Ctrl+C to shutdown the network.
+
+The command output shows the addresses of the CCF nodes where commands may be submitted (eg, in this case, via ``curl https://127.251.192.205:36981/...``). The output and error logs of each node can be found in the node-specific directory in the workspace (eg, ``workspace/test_network_0/err`` is node 0's stderr).
+
+Authentication
+--------------
+
+When establishing a TLS connection with a CCF service both the service and client must prove their identity.
+
+The service identity is created at startup. The initial node generates a fresh private key which exists solely within the service's enclaves. A certificate of the corresponding public key is emitted (``networkcert.pem``) and used by all subsequent connections to confirm they are communicating with the intended service.
+
+Each member and user is identified by the cert with which they were registered with the service, either at genesis or in a subsequent ``new_member`` or ``new_user`` governance proposal. Access to the corresponding private key allows a client to submit commands as this member or user. For this test network these are all freshly generated and stored in the same common workspace for easy access. In a real deployment only the certificates would be shared; the private keys would be distributed and remain confidential.
+
+When using curl the server's identity is provided by ``--cacert`` and the client identity by ``--cert`` and ``--key``. Resources under the ``/gov`` path require member identities, while those under ``/app`` require user identities.
```

Comment:
Nitpick: "while those under `/app` _typically_ require user identities" (as we can install handlers without authz, and will make use of those more and more in the future).

Reply:
I hummed and hawed about this a bit, because I expect it to change soon (with frontend abolition, the auth requirement is entirely per-endpoint, and apps can probably add user, unauth'd, and member endpoints?). So I don't want to be too specific about the current options. But I think your wording is accurate for now, so I'll add it.